packages/office-addin-telemetry/src/officeAddinTelemetry.ts

```diff
@@ -0,0 +1,301 @@
+import * as appInsights from "applicationinsights";
+import * as chalk from "chalk";
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+import * as readLine from "readline-sync";
+export enum telemetryType {
+  applicationinsights = "applicationInsights",
+  // OtelJs = "OtelJs" - Not yet implemented
+}
+
+const telemetryJsonFilePath: string = path.join(os.homedir(), "/officeAddinTelemetry.json");
+
+/**
+ * Telemetry object necesary for initialization of telemetry package
+ * @member groupName Event name sent to telemetry structure
+ * @member instrumentationKey Instrumentation key for telemetry resource
+ * @member promptQuestion Question displayed to User over opt-in for telemetry
+ * @member telemetryEnabled User's response to the prompt for telemetry
+ * @member telemetryType Telemetry infrastructure to send data
+ * @member testData Allows user to run program without sending actuall data
+ */
+export interface ITelemetryObject {
+  groupName: string;
+  instrumentationKey: string;
+  promptQuestion: string;
+  raisePrompt: boolean;
+  telemetryEnabled: boolean;
+  telemetryJsonFilePath: string;
+  telemetryType: telemetryType;
+  testData: boolean;
+}
+
+/**
+ * Creates and initializes member variables while prompting user for telemetry collection when necessary
+ * @param telemetryObject
+ */
+export class OfficeAddinTelemetry {
+  private telemetryClient = appInsights.defaultClient;
+  private eventsSent: number = 0;
+  private exceptionsSent: number = 0;
+  private telemetryObject: ITelemetryObject;
+
+  constructor(telemetryObj: ITelemetryObject) {
+    try {
+      this.telemetryObject = telemetryObj;
+      if (this.telemetryObject.telemetryJsonFilePath === undefined){
+        this.telemetryObject.telemetryJsonFilePath = telemetryJsonFilePath;
+      }
+
+      if (!this.telemetryObject.testData && this.telemetryObject.raisePrompt && promptForTelemetry(this.telemetryObject.groupName)) {
+        this.telemetryOptIn();
+      } else {
+        const telemetryJsonData = readTelemetryJsonData(this.telemetryObject.telemetryJsonFilePath);
+        if (telemetryJsonData) {
+          if (!groupNameExists(telemetryJsonData, this.telemetryObject.groupName)) {
+            telemetryJsonData.telemetryInstances[this.telemetryObject.groupName] = this.telemetryObject.telemetryEnabled;
+            writeTelemetryJsonData(telemetryJsonData, this.telemetryObject.telemetryJsonFilePath);
+          }
+        } else {
+          writeNewTelemetryJsonFile(this.telemetryObject.groupName, this.telemetryObject.telemetryEnabled, this.telemetryObject.telemetryJsonFilePath);
+        }
+      }
+
+      if (this.telemetryObject.instrumentationKey === undefined) {
```

Comment:
I think this throw should go right at the beginning of the constructor so we don’t waste time doing other stuff if the instrumentation key isn’t set